client/lib/sites-list/index.js

```diff
@@ -16,7 +22,15 @@ module.exports = function() {
 				case 'RECEIVE_DELETED_SITE':
 					_sites.removeSite( action.site );
 					break;
-
+				case InvitesActionTypes.INVITE_ACCEPTED:
+					let sites = clone( _sites.get() );
+					if ( sites.length === 0 ) {
+						action.invite.site.primary = true;
+					}
+					sites.push( action.invite.site );
```

Comment:
We are already doing that, `_sites.update` calls `createSiteObject` on each new site that we pass on `sites`